crates/primitives/src/utils/mod.rs

```diff
@@ -16,6 +16,8 @@ pub use units::{
 cfg_if! {
     if #[cfg(all(feature = "asm-keccak", not(miri)))] {
         use keccak_asm::Digest as _;
+    } else if #[cfg(all(feature = "sha3-keccak", not(miri)))] {
+        use sha3::Digest as _;
```

Comment:
what happens if both asm-keccak and sha3.keccak are enabled? does it compile?

Reply:
Yep this is checked on CI I believe:

https://github.com/alloy-rs/core/blob/022c63db87b6bca348dd8e55f86c94a614b99cb6/.github/workflows/ci.yml#L100-L111